File: omrchecker-js/packages/core/src/processors/detection/bubbles_threshold/fileRunner.ts
/**
 * Bubbles threshold file runner.
 *
 * TypeScript port of src/processors/detection/bubbles_threshold/file_runner.py
 * Extends FieldTypeFileLevelRunner for bubble field processing.
 */

import { FieldDetectionType } from '../../constants';
import { FieldTypeFileLevelRunner } from '../base/fileRunner';
import { BubblesThresholdDetectionPass } from './detectionPass';
import { BubblesThresholdInterpretationPass } from './interpretationPass';
import type { TuningConfig } from '../base/commonPass';
import { DetectionRepository } from '../../repositories/DetectionRepository';

/**
 * File runner for bubbles threshold detection and interpretation.
 *
 * Instantiates BubblesThresholdDetectionPass and BubblesThresholdInterpretationPass
 * and coordinates them for processing bubble fields.
 */
export class BubblesThresholdFileRunner extends FieldTypeFileLevelRunner {
  public repository: DetectionRepository;

  constructor(tuningConfig: TuningConfig, repository: DetectionRepository) {
    const fieldDetectionType = FieldDetectionType.BUBBLES_THRESHOLD;
    const detectionPass = new BubblesThresholdDetectionPass(
      tuningConfig,
      fieldDetectionType,
      repository
    );
    const interpretationPass = new BubblesThresholdInterpretationPass(
      tuningConfig,
      fieldDetectionType,
      repository
    );

    super(tuningConfig, fieldDetectionType, detectionPass, interpretationPass);
    this.repository = repository;
  }
}

File: src/processors/detection/bubbles_threshold/file_runner.py
from src.processors.constants import FieldDetectionType
from src.processors.detection.base.file_runner import FieldTypeFileLevelRunner
from src.processors.detection.bubbles_threshold.detection_pass import (
    BubblesThresholdDetectionPass,
)
from src.processors.detection.bubbles_threshold.interpretation_pass import (
    BubblesThresholdInterpretationPass,
)
from src.processors.repositories.detection_repository import DetectionRepository


class BubblesThresholdFileRunner(FieldTypeFileLevelRunner):
    def __init__(self, tuning_config, repository: DetectionRepository) -> None:
        field_detection_type = FieldDetectionType.BUBBLES_THRESHOLD
        detection_pass = BubblesThresholdDetectionPass(
            tuning_config, field_detection_type, repository=repository
        )
        interpretation_pass = BubblesThresholdInterpretationPass(
            tuning_config, field_detection_type, repository=repository
        )
        super().__init__(
            tuning_config, field_detection_type, detection_pass, interpretation_pass
        )
        self.repository = repository
